docs/matrix_format_spec.md

```diff
@@ -0,0 +1,101 @@
+stores contain groups and arrays. Groups can contain other groups and arrays.
+Arrays are stored chunked and compressed. The file names and directory
+structure of the zarr store convey the group and chunk structure.
+
+# Groups and Arrays
+
+Expression values and metadata are stored in seven arrays. These arrays are
+stored within a single zarr group called `expression_matrix`.
+
+- expression - The expression values themselves. Rows correspond to cells and
+  columns to genes. These must represent raw counts. Any normalization is
+  deferred, but metadata that enables normalization should be included in the
+  metadata arrays defined below.
+  - shape: `(n_cells, n_genes)`
+  - compressor: `Blosc(cname='lz4', clevel=5, shuffle=SHUFFLE, blocksize=0)`
+  - chunk shape: `(10000, n_genes)`
+  - dtype: `int32`
+
+- cell_id - Identifiers for each cell. The means of identifying a cell varies
+  by experiment type. For example, in droplet data this is a barcode; in
+  well-based assays it is a well identifier. The `ith` element of this array
+  identifies the `ith` row of the `expression` array. Elements in this array
+  must be unique.
+  - shape: `(n_cells,)`
+  - compressor: `Blosc(cname='lz4', clevel=5, shuffle=SHUFFLE, blocksize=0)`
+  - chunk shape: `(10000,)`
+  - dtype: `U40`
+
+- gene_id - Identifiers for each gene. The `ith` element of this array
+  identifies the gene at the `jth` column of the `expression` array. The
+  `gene_id` array must be identical for all expression data produced by HCA
+  secondary analysis pipelines. Elements in this array must be unique.
+  - shape: `(n_genes,)`
+  - compressor: `Blosc(cname='lz4', clevel=5, shuffle=SHUFFLE, blocksize=0)`
+  - chunk shape: `(n_genes,)`
+  - dtype: `U40`
+
+- cell_metadata - Numeric metadata values associated with each cell. These
+  vary by experiment type but should be as consistent as possible. These
+  include quality control values as well as other computed values that aid in
+  cell filtering and selection.
+  - shape: `(n_cells, n_cell_metadata_values)`
+  - compressor: `Blosc(cname='lz4', clevel=5, shuffle=SHUFFLE, blocksize=0)`
+  - chunk shape: `(10000, n_cell_metadata_values)`
+  - dtype: `float32`
+
+- gene_metadata - Numeric metadata values associated with each gene. Like
+  `cell_metadata` these can vary by experiment type, but care should be taken
+  to maintain consistency. 
+  - shape: `(n_gene_metadata_values, n_genes)`
+  - compressor: `Blosc(cname='lz4', clevel=5, shuffle=SHUFFLE, blocksize=0)`
+  - chunk shape: `(n_gene_metadata_values, n_genes)`
+  - dtype: `float32`
+
+- cell_metadata_name - Field names for `cell_metadata`. These must be unique.
+  - shape: `(n_cell_metadata_values,)`
+  - compressor: `Blosc(cname='lz4', clevel=5, shuffle=SHUFFLE, blocksize=0)`
+  - chunk shape: `(n_cell_metadata_values,)`
+  - dtype: `U40`
+
+- gene_metadata_name - Field names for `gene_metadata`. These must be unique.
```

Comment:
For consistency with gene_id - _Elements in this array must be unique._ ?

Reply:
👍